### PR TITLE
misc(core): add drop marker whenever NaN is encountered during ingestion

### PR DESCRIPTION
**Pull request by sherali42**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?

- Current Behavior: Whenever `NaN` is encountered, counter drop is not marked. But counter drop is marked when the subsequent value is lower than the previous non NaN value. But this breaks if NaN occurs at the chunk boundary.

- Changes: As NaN represents end of time-series, A counter drop is marked Whenever an NaN is ingested. For value comparison purposes NaN is considered as `0`. This fixes chunk boundary case as well.

thanks @vishramachandran for contributing to the PR

### CODE REVIEW

**Review by vishramachandran — APPROVED** [review]

File: spark-jobs/src/test/scala/filodb/downsampler/DownsamplerMainSpec.scala [hunk 1]
```diff
@@ -323,6 +324,57 @@ class DownsamplerMainSpec extends AnyFunSpec with Matchers with BeforeAndAfterAl
...
+      Seq(74372921500L, 7d, 9d, LongHistogram(bucketScheme, Array(1L, 7, 9)), histNameNaN, seriesTagsNaN),
+      Seq(74372922000L, Double.NaN, Double.NaN, LongHistogram(bucketScheme, Array(0L, 0, 0)), histNameNaN, seriesTagsNaN),
+      Seq(74372922500L, 4d, 1d, LongHistogram(bucketScheme, Array(0L, 1, 1)), histNameNaN, seriesTagsNaN),
+
+      Seq(74372981000L, 17d, 21d, LongHistogram(bucketScheme, Array(2L, 16, 21)), histNameNaN, seriesTagsNaN),
+      Seq(74372981500L, 1d, 1d, LongHistogram(bucketScheme, Array(0L, 1, 1)), histNameNaN, seriesTagsNaN),
+      Seq(74372982000L, 15d, 15d, LongHistogram(bucketScheme, Array(0L, 15, 15)), histNameNaN, seriesTagsNaN),
+
+      Seq(74373041000L, 18d, 19d, LongHistogram(bucketScheme, Array(1L, 16, 19)), histNameNaN, seriesTagsNaN),
+      Seq(74373042000L, 20d, 25d, LongHistogram(bucketScheme, Array(4L, 20, 25)), histNameNaN, seriesTagsNaN)
```
[the remark this answers is not in the document]
Comment (sherali42) [reply spark-jobs/src/test/scala/filodb/downsampler/DownsamplerMainSpec.scala]: added `NaN` at the end
